File: Week2/readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>
int count_letters(string n);
int count_words(string n);
int count_sentences(string n);

int main(void)
{
    string text = get_string("Test: ");
    int letters = count_letters(text);     // 字母總數
    int words = count_words(text);         // 單字總數
    int sentences = count_sentences(text); // 句子總數

    double L = (double) letters / words * 100;
    double S = (double) sentences / words * 100;

    double index = 0.0588 * L - 0.296 * S - 15.8;
    int grade = (int) round(index);
    // printf("%f,%f,%d\n",L,S,index);
    if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (grade >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", grade);
    }
    return 0;
}

int count_letters(string text)
{
    int count = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (isalpha(text[i]))
        {
            count++;
        }
    }
    return count;
}

int count_words(string text)
{
    if (strlen(text) == 0)
    {
        return 0;
    }
    int count = 1; // 第一個單字前沒有空白
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (text[i] == ' ')
        {
            count++;
        }
    }
    return count;
}
int count_sentences(string text)
{
    int count = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            count++;
        }
    }
    return count;
}

File: Week6/readability/readability.py
text = input("Text:")


def count_letters(TText):
    count = 0
    for i in TText:
        if i.isalpha():
            count += 1
    return count


def count_words(TText):
    count = 1
    for i in TText:
        if i == " ":
            count += 1
    return count


def count_sentences(TText):
    count = 0
    for i in TText:
        if i == "." or i == "!" or i == "?":
            count += 1
    return count


letters = count_letters(text)
words = count_words(text)
sentences = count_sentences(text)

L = letters / words * 100
S = sentences / words * 100

index = 0.0588 * L - 0.296 * S - 15.8
grade = round(index)

if grade < 1:
    print("Before Grade 1")
elif grade >= 16:
    print("Grade 16+")
else:
    print(f"Grade {grade}")
